### PR TITLE
Query: Relational: Adds composite key support to nav. prop translation.

### PR DESCRIPTION
**Pull request by anpete**

### CODE REVIEW

**Comment by anpete** [comment]
@maumar

File: src/EntityFramework.Relational/Query/ExpressionVisitors/SqlTranslatingExpressionVisitor.cs [hunk 1]
```diff
@@ -151,26 +151,47 @@ private static Expression UnfoldStructuralComparison(ExpressionType expressionTy
...
             var leftExpressions = leftConstantExpression?.Value as Expression[];
+            var rightConstantExpression = binaryExpression?.Right as ConstantExpression;
+            var rightExpressions = rightConstantExpression?.Value as Expression[];
 
             if (leftExpressions != null)
             {
-                var rightConstantExpression = binaryExpression.Right as ConstantExpression;
-
-                var rightExpressions = rightConstantExpression?.Value as Expression[];
+                if (rightConstantExpression != null
```
Comment (maumar) [inline src/EntityFramework.Relational/Query/ExpressionVisitors/SqlTranslatingExpressionVisitor.cs]: collapse into one if statement?

File: src/EntityFramework.Relational/Query/ExpressionVisitors/SqlTranslatingExpressionVisitor.cs [hunk 2]
```diff
@@ -151,26 +151,47 @@ private static Expression UnfoldStructuralComparison(ExpressionType expressionTy
...
+                            .Repeat<Expression>(rightConstantExpression, leftExpressions.Length)
+                            .ToArray();
+                }
+            }
 
-                if (rightExpressions != null
-                    && leftExpressions.Length == rightExpressions.Length)
+            if (rightExpressions != null)
+            {
+                if (leftConstantExpression != null
```
Comment (maumar) [inline src/EntityFramework.Relational/Query/ExpressionVisitors/SqlTranslatingExpressionVisitor.cs]: same as above

**Comment by maumar** [comment]
:shipit:

File: src/EntityFramework.Relational/Query/ExpressionVisitors/SqlTranslatingExpressionVisitor.cs (hunk 1 above)
Comment (anpete) [inline src/EntityFramework.Relational/Query/ExpressionVisitors/SqlTranslatingExpressionVisitor.cs]: Fixed thanks.

File: src/EntityFramework.Relational/Query/ExpressionVisitors/SqlTranslatingExpressionVisitor.cs (hunk 2 above)
Comment (anpete) [inline src/EntityFramework.Relational/Query/ExpressionVisitors/SqlTranslatingExpressionVisitor.cs]: Fixed thanks.

**Pull request merged**